pkg/apis/v1beta1/ec2nodeclass_validation_cel_test.go

```diff
@@ -0,0 +1,576 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1_test
+
+import (
+	"strings"
+
+	"github.com/Pallinder/go-randomdata"
+	"github.com/imdario/mergo"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+	"github.com/aws/karpenter/pkg/apis/v1beta1"
+	"github.com/aws/karpenter/pkg/test"
+)
+
+var _ = Describe("CEL/Validation", func() {
+	var nc *v1beta1.EC2NodeClass
+
+	BeforeEach(func() {
+		v := lo.Must1(awsEnv.VersionProvider.Get(ctx))
+		if v == "1.23" || v == "1.24" {
```

Comment:
Can we just check this from the environment in the same way that we check it in `karpenter-core`? Not sure that we need to utilize the version provider here